### PR TITLE
[BEAM-463] [BEAM-464] [BEAM-466] BoundedHeapCoder, HolderCoder, QuantileStateCoder -> StandardCoder

### PR DESCRIPTION
**Pull request by aviemzur**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by aviemzur** [comment]
R: @lukecwik

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Top.java [hunk 1]
```diff
@@ -550,6 +551,11 @@ public void encode(
     }
 
     @Override
+    public List<? extends Coder<?>> getCoderArguments() {
+      return null;
```
Comment (kennknowles) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Top.java]: This should return a `Coder<T>` - the coder arguments should correspond to the type arguments.

**Comment by aaltay** [comment]
Is this ready for review again?

**Comment by aviemzur** [comment]
Closing this PR for now. Will keep the branch open if the ticket becomes relevant again.

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Top.java [hunk 2]
```diff
@@ -550,6 +552,12 @@ public void encode(
     }
 
     @Override
+    public List<? extends Coder<?>> getCoderArguments() {
+      Coder<?> elementCoder = listCoder.getCoderArguments().get(0);
```
Comment (kennknowles) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Top.java]: Notice that you could just keep `elementCoder` in a field (as well as caching `listCoder`) on line 536. No need to pull it out here.

**Comment by kennknowles** [comment]
I think this is actually about ready. It will have some revision again when moving to the Runner API but it is a good setup.

**Pull request closed without merging**